presto-parser/src/main/java/com/facebook/presto/sql/ExpressionFormatter.java

```diff
@@ -566,6 +570,43 @@ static String formatSortItems(List<SortItem> sortItems, boolean unmangleNames)
                 .iterator());
     }
 
+    static String formatGroupBy(List<GroupingElement> groupingElements)
+    {
+        checkState(groupingElements.stream()
```

Comment:
It should be trivial to support formatting in the case of multiple grouping sets, no?